evasquez26/angular4-startup
# Create quiz and quiz questions

## Story / Task

[ONE-115](https://bixlabs.atlassian.net/browse/ONE-115), [ONE-116](https://bixlabs.atlassian.net/browse/ONE-116), [ONE-56](https://bixlabs.atlassian.net/browse/ONE-56) and [ONE-200](https://bixlabs.atlassian.net/browse/ONE-200)

## What this PR does?

Allows an owner of a course to add quizzes to the lessons and questions to the quizzes. DISCLAIMER: Design has not being applied yet, so the form looks kind of messy

### Before 

Owner of a course could not add quizzes to lessons

### After

An owner of a course can now add quizzes to lessons and questions to the quizzes

### How to test / reproduce

- Login into the app as a teacher
- Create a course
- Observe that there's a tab of lessons
- Add a new lesson to the course
- Observe that there's a button that will allow you add a single quiz to the lesson
- Add the quiz to the lesson
- Observe that there's a button for adding questions to the quiz
- Clicking the button should present you with different options (Multiple choice question or written answer)
- Choosing written answer should display a single markdown input and a tags input
- Leaving fields on blank and clicking the save button should show an error on the required fields
- Filling the information correctly and clicking the save button should save the question and list it under the corresponding lesson
- Choosing multiple choice question should display additional options
- Two radio buttons to specify if the answers will be text or images
- Choosing text should display two tag inputs for correct and incorrect answers respectively
- Choosing images should display two input files for  correct and incorrect answers respectively
- Choosing multiple files should list them on the corresponding selected field
- Clicking the save button should save the corresponding data and retrieving it correctly
- Observe that there's also a delete button for the question if you have more than 1
- Clicking the delete button should delete the question, except the last one

### TODOs

N/A

### Related PRs

N/A

## Migrations

NO

Thanks!
